pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/SegmentFetcherFactory.java

```diff
@@ -118,6 +125,47 @@ public static void fetchSegmentToLocal(String uri, File dest)
     fetchSegmentToLocal(new URI(uri), dest);
   }
 
+  /**
+   * Fetches a segment from a given URI and untar the segment file to the dest dir (i.e., tableDataDir + segmentName).
+   */
+  public static void fetchAndUntarSegmentToLocal(String uri, File tableDataDir, String segmentName)
```

Comment:
Why do we need this method? There is already an `untarAndMoveSegment()` within the `RealtimeTableDataManager` class. We should be able to use that

Reply:
Done. Good catch. Use the existing downloadSegmentFromDeepstore() method which does the same function.